### PR TITLE
Fix for issues 254 & 214

### PR DESCRIPTION
**Pull request by krishollenbeck**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Fixes for the following tickets

**Hierarchy Issue:**
> Hierarchy (enterprise) control now supports a submenu and disabled menu items in the actions menu. Need to add an example of that functionality to the NG version. Also need to update SohoHierarchyAction with new properties supported in the enterprise version.

**RenderLoop Issue:**
> Currently renderLoop is initialized via Sohoxi.js. If we wrap this with an Angular wrapper (possibly a service), we could initialize it via app.module. For example..

**Related github/jira issue (required)**:

1. https://github.com/infor-design/enterprise-ng/issues/254
2. https://github.com/infor-design/enterprise-ng/issues/214

**Steps necessary to review your pull request (required)**:

**Hierarchy Issue:**
1. Apply changes from PR
2. Navigate to localhost:4203/hierarchy
3. Click the actions menu button on "Patricia Clark"
4. There should be a disabled "Change Pay Rate" and "Options" should have a sub menu

**RenderLoop Issue:**
1. Apply changes from PR
2. Navigate to the home page (or any page)
3. At the top of the page the current RenderLoop count should display on load.
4. Click stop to stop the loop & start to start the loop.
5. Comment out `this.renderLoop.start();` in **app.component.ts**. Notice the loop will not start now. This allows the Angular app to control when the loop should be initialized.

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import {
+  AfterViewChecked,
   Component,
-  HostBinding,
+  HostBinding, OnInit,
```
Comment (pwpatton) [inline src/app/app.component.ts]: put on own line

File: src/index.html [hunk 2]
```diff
@@ -5,6 +5,11 @@
   <title>SohoXI - Angular Components</title>
   <base href="/">
 
+  <!-- prevent render loop from initializing on load -->
```
Comment (bthharper) [inline src/index.html]: Is this something we should all look at doing?

Comment (krishollenbeck) [reply src/index.html]: Yeah, before [Eds change](https://github.com/infor-design/enterprise/pull/1098) renderLoop initialized on load. He has now changed that to be optional via a config setting.

`if (this.settings.noAutoStart !== true) {
      this.start();
    }
`

We can set this in index.html before sohoxi.js loads . And then start it later manually in our ### app.component _ngOnInit()_ There is a requestAnimationFrame in the renderLoop that was causing changeDetection to fire. The renderLoop service allows us to call that outside of the Angular zone.

Comment (krishollenbeck) [reply src/index.html]: @bthharper I can add a note to the change log. Probably should have done that.

Comment (pwpatton) [reply src/index.html]: I think this can wait for ep-ng of 5.x.

Comment (bthharper) [reply src/index.html]: Okay - might be worth creating the new CHANGELOG section.

**Pull request merged**